tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -70,12 +70,22 @@ namespace tiledb::api {
 
 // Forward declarations
 uint64_t compute_tile_extent_based_on_file_size(uint64_t file_size);
-std::string libmagic_get_mime(void* data, uint64_t size);
-std::string libmagic_get_mime_encoding(void* data, uint64_t size);
-bool libmagic_file_is_compressed(void* data, uint64_t size);
-void read_file_header(
-    tiledb::sm::VFS& vfs, const char* uri, std::vector<char>& header);
+
+static std::string get_mime(Logger& log, tiledb_mime_type_t mime) {
```

Comment:
NIT: It'a a bit odd for this function to require a logger argument, can't we just log on the caller side if needed?